File: Heat/Models/WarehouseMovement.vb
Imports System.ComponentModel.DataAnnotations

Namespace Models

    Public Class WarehouseMovement

        Property ID As Integer

        'EF by-configuration foreign key
        Property ProductID As Integer
        Property Product As Product

        Property Quantity As Double
        <DataType(DataType.Date), DisplayFormat(dataformatstring:="{0:yyyy-MM-dd}", applyFormatInEditMode:=True)> _
        Property ExecDate As Date
        Property Note As String

        Property SourceID As Integer
        Property Source As Warehouse

        Property DestinationID As Integer
        Property Destination As Warehouse

        Property CausalWarehouseID As Integer
        Property CausalWarehouse As CausalWarehouse

    End Class
End Namespace

File: Heat.ConvertedToC#/Models/WarehouseMovement.cs
using System.ComponentModel.DataAnnotations;

namespace Heat.Models
{

    public class WarehouseMovement
	{

		public int ID { get; set; }

		//EF by-configuration foreign key
		public int ProductID { get; set; }
		public Product Product { get; set; }

		public double Quantity { get; set; }
		[DataType(DataType.Date), DisplayFormat(DataFormatString = "{0:yyyy-MM-dd}", ApplyFormatInEditMode = true)]
		public System.DateTime ExecDate { get; set; }
		public string Note { get; set; }

		public int SourceID { get; set; }
		public Warehouse Source { get; set; }

		public int DestinationID { get; set; }
		public Warehouse Destination { get; set; }

		public int CausalWarehouseID { get; set; }
		public CausalWarehouse CausalWarehouse { get; set; }

	}
}
